src/lib/exports.ts

```diff
@@ -403,3 +426,350 @@ export const exportNoteAsPdfFile = async (
     })
   }
 }
+
+export function getExportAsMenuItems(
+  folderName: string,
+  folderPathname: string,
+  recursive: boolean,
+  startExport: (
+    folderName: string,
+    folderPathname: string,
+    exportType: string,
+    recursive: boolean
+  ) => void
+): MenuItemConstructorOptions[] {
+  return [
+    {
+      type: 'normal',
+      label: 'Export as Markdown (.md)',
+      click: () => startExport(folderName, folderPathname, 'md', recursive),
+    },
+    {
+      type: 'normal',
+      label: 'Export as HTML (.html)',
+      click: () => startExport(folderName, folderPathname, 'html', recursive),
+    },
+    {
+      type: 'normal',
+      label: 'Export as PDF (.pdf)',
+      click: () => startExport(folderName, folderPathname, 'pdf', recursive),
+    },
+  ]
+}
+
+function getPercentage(
+  value: number,
+  maxValue: number,
+  exportProgressMaxValue = 100
+) {
+  if (maxValue === 0) {
+    return 0
+  }
+  return Math.floor((value / maxValue) * exportProgressMaxValue)
+}
+
+async function writeNoteData(
+  noteData: Buffer | string,
+  location: string,
+  exportErrors: string[]
+) {
+  try {
+    await writeFile(location, noteData)
+  } catch (err) {
+    exportErrors.push(
+      `Cannot export: '${location.substring(
+        location.lastIndexOf('/')
+      )}', reason: ${err}`
+    )
+  }
+}
+
+async function getAttachmentData(storage: NoteStorage, src: string) {
```

Comment:
1. Please throw error rather than returning reject
2. Rename the function. Its name should explain what it does.
3. Reconsider that we really need this function.

Reply:
Refactored/removed.